qiita_ware/ebi.py

```diff
@@ -143,33 +126,76 @@ def __init__(self, preprocessed_data_id, action):
             self.new_investigation_type = it
         else:
             # This should never happen
-            raise ValueError("Unrecognized investigation type: '%s'. This "
-                             "term is neither one of the official terms nor "
-                             "one of the user-defined terms in the ENA "
-                             "ontology")
-
-        self.ebi_dir = join(qiita_config.working_dir, 'ebi_submission_',
-                            str(preprocessed_data_id))
-        self.sequence_files = []
-        self.study_xml_fp = None
-        self.sample_xml_fp = None
-        self.experiment_xml_fp = None
-        self.run_xml_fp = None
-        self.submission_xml_fp = None
+            error_msgs.append("Unrecognized investigation type: '%s'. This "
+                              "term is neither one of the official terms nor "
+                              "one of the user-defined terms in the ENA "
+                              "ontology." % it)
+
+        self.ebi_dir = join(qiita_config.working_dir,
+                            'ebi_submission_%d' % preprocessed_data_id)
+        get_output_fp = partial(join, self.ebi_dir, 'xml_dir')
+        self.xml_dir = get_output_fp('')
+        self.study_xml_fp = get_output_fp('study.xml')
+        self.sample_xml_fp = get_output_fp('sample.xml')
+        self.experiment_xml_fp = get_output_fp('experiment.xml')
+        self.run_xml_fp = get_output_fp('run.xml')
+        self.submission_xml_fp = get_output_fp('submission.xml')
         self.pmids = s.pmids
 
-        # dicts that map investigation_type to library attributes
-        lib_strategies = {'metagenome': 'POOLCLONE',
-                          'mimarks-survey': 'AMPLICON'}
-        lib_selections = {'mimarks-survey': 'PCR'}
-        lib_sources = {}
-
-        self.library_strategy = lib_strategies.get(key, "OTHER")
-        self.library_source = lib_sources.get(key, "METAGENOMIC")
-        self.library_selection = lib_selections.get(key, "unspecified")
-
-        # This will hold the submission's samples, keyed by the sample name
+        # ignore data type for the restrictions as we assume that this was
```

Comment:
This comment is not accurate because you're actually checking the data type on line 153.

Anyways, I don't think the logic here is correct. In here we should be checking the columns required by EBI. In the current code, you're checking all the required columns, for multiple reasons - however I think that the correct thing to do is checking only the EBI columns.

Also, you don't need to actually check the data type - the templates have a property that will return the columns that are required for their datatype:
```python
In [11]: st = SampleTemplate(1)

In [12]: st.columns_restrictions
Out[12]: 
{'EBI': Restriction(columns={'physical_specimen_location': 'varchar', 'collection_timestamp': 'timestamp', 'scientific_name': 'varchar', 'taxon_id': 'integer'}, error_msg='EBI submission disabled'),
 'qiita_main': Restriction(columns={'description': 'varchar', 'physical_specimen_remaining': 'bool', 'dna_extracted': 'bool', 'longitude': 'float8', 'sample_type': 'varchar', 'latitude': 'float8', 'host_subject_id': 'varchar'}, error_msg='Processed data approval disabled')}

In [13]: pt = PrepTemplate(1)

In [14]: pt.columns_restrictions
Out[14]: 
{'EBI': Restriction(columns={'center_name': 'varchar', 'platform': 'varchar', 'primer': 'varchar', 'experiment_design_description': 'varchar', 'library_construction_protocol': 'varchar'}, error_msg='EBI submission disabled'),
 'demultiplex': Restriction(columns={'primer': 'varchar', 'barcode': 'varchar'}, error_msg='Demultiplexing disabled. You will not be able to preprocess your raw data'),
 'demultiplex_multiple': Restriction(columns={'primer': 'varchar', 'barcode': 'varchar', 'run_prefix': 'varchar'}, error_msg='Demultiplexing with multiple input files disabled. If your raw data includes multiple raw input files, you will not be able to preprocess your raw data')}
```

Given this, I think the correct path here is:
```python
restrictions = self.sample_template.columns_restrictions['EBI'].columns.keys()
restrictions.extend(self.prep_template.columns_restrictions['EBI'].columns.keys()
```